static/js/controllers/user-language-modal.js

```diff
@@ -0,0 +1,64 @@
+  var inboxServices = angular.module('inboxServices');
+
+  inboxServices.factory('UserLanguageModal', ['$uibModal',
+    function($uibModal) {
+      return function(enabledLocales) {
+        var modalInstance = $uibModal.open({
+          templateUrl: 'templates/modals/user_language.html',
+          controller: 'UserLanguageModalCtrl',
+          resolve: {
+            enabledLocales: function () {
+              return enabledLocales;
+            }
+          }
+        });
+        return modalInstance.result;
+      };
+    }
+  ]);
+
+angular.module('inboxControllers').controller('UserLanguageModalCtrl',
+  ['enabledLocales', '$log', '$q', '$scope', 'Session', 'SetLanguage', '$translate', '$uibModalInstance', 'UpdateUser',
+  function(enabledLocales, $log, $q, $scope, Session, SetLanguage, $translate, $uibModalInstance, UpdateUser) {
```

Comment:
I find enabledLocales quite confusing. I take it it's being injected from the resolve property in the UserLanguageModal factory? Can you think of any other way to pass this in? Maybe it could call a function called `$scope.init` or something? Or is this the way `uibMobal` expects it to be?

I think the confusing thing is usually we inject services, factories, constants, etc. This is the first time I've seen state being injected.

Another solution is to remove the `resolve` bit from above, and have the controller query a service for all enabled locales. That would reduce the coupling significantly as the Modal will work without dependencies on other controllers.